cs/test/expressions/SerializeCB.expressions

```diff
@@ -327,26 +327,221 @@
                 .Constant<System.UInt16>(40),
                 .Constant<Bond.Metadata>(_nestedVector))
         };
-        .If ($Example.b != .Default(System.ArraySegment`1[System.Byte])) {
+        .Block(System.ArraySegment`1[System.Byte] $convertedBlob) {
+            $convertedBlob = $Example.b;
+            .If ($convertedBlob != .Default(System.ArraySegment`1[System.Byte])) {
+                .Block() {
+                    .Call $writer.WriteFieldBegin(
+                        .Constant<Bond.BondDataType>(BT_LIST),
+                        .Constant<System.UInt16>(50),
+                        .Constant<Bond.Metadata>(b));
+                    .Block(
+                        System.ArraySegment`1[System.Byte] $arraySegment,
+                        System.Int32 $count,
+                        System.Int32 $index,
+                        System.Int32 $end) {
+                        $arraySegment = $Example.b;
```

Comment:
Is there any simplification that can be done to get rid of this older optimization for a similar problem?

Reply:
Investigating this.

Reply:
Removing this adds quite a bit of complexity between the code paths where blobs require conversion and where they don't. Not sure it is worth addressing this unless expression optimizer isn't capable of the appropriate peephole optimizations. Thoughts?

Reply:
Thanks for looking. If it's not super simple, I'd skip the removal.